presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -291,6 +294,33 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
             // given subquery is already appended
             return subPlan;
         }
+        switch (quantifiedComparison.getComparisonType()) {
+            case EQUAL:
+                switch (quantifiedComparison.getQuantifier()) {
+                    case ANY:
+                    case SOME:
+                        // A = ANY B <=> A IN B
+                        InPredicate inPredicate = new InPredicate(quantifiedComparison.getValue(), quantifiedComparison.getSubquery());
+                        subPlan.getTranslations().addIntermediateMapping(quantifiedComparison, inPredicate);
+                        return appendInPredicateApplyNode(subPlan, inPredicate, correlationAllowed);
+                }
+                break;
+
+            case NOT_EQUAL:
+                switch (quantifiedComparison.getQuantifier()) {
+                    case ALL:
+                        // A <> ALL B <=> !(A IN B) <=> !(A = ANY B)
+                        QuantifiedComparisonExpression rewrittenAny = new QuantifiedComparisonExpression(
+                                ComparisonExpressionType.EQUAL,
+                                Quantifier.ANY,
+                                quantifiedComparison.getValue(),
+                                quantifiedComparison.getSubquery());
+                        Expression notAny = new NotExpression(rewrittenAny);
+                        subPlan.getTranslations().addIntermediateMapping(quantifiedComparison, notAny);
+                        return appendQuantifiedComparisonApplyNode(subPlan, rewrittenAny, correlationAllowed);
```

Comment:
Yes, it's the intended implementation. We know that `A <> ALL B` is equivalent to `NOT (A = ANY B)`, so I add a rewrite for the initial `quantifiedComparison` to `notAny`. However, we still need to plan `ANY B` part, so I call the same function to plan `rewrittenAny`.
Let me know if there's any way I can make it more clear in the code.
Also, current tests, including integration tests, should catch any issues with this code.